src/types.ts

```diff
@@ -35,13 +35,18 @@ export type DeepImmutable<T> = T extends Primitive
   ? DeepImmutableMap<K, V>
   : DeepImmutableObject<T>
 
-export type Reducer<TState, TAction> = (
+export type Handler<TState, TAction> = (
   prevState: DeepImmutable<TState>,
   action: TAction
 ) => DeepImmutable<TState> | TState
 
+export type Reducer<TState, TAction> = (
+  state: DeepImmutable<TState> | undefined,
+  action: TAction
+) => DeepImmutable<TState> | TState
+
 export type ActionType<
-  T extends ActionCreator<AnyAction> | Reducer<any, Action<any>>
+  T extends ActionCreator<AnyAction> | Handler<any, Action<any>>
```

Comment:
The `ActionType` helper should get a reducer or action creator and returns its corresponding action. So the previous version was correct.

```suggestion
  T extends ActionCreator<AnyAction> | Reducer<any, Action<any>>
```